### PR TITLE
Added support for new message api endpoints

### PR DESCRIPTION
**Pull request by petergarbers**

### CODE REVIEW

File: zep_python/message/client.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+        if limit is not None and cursor is not None:
+            if (
+                not isinstance(limit, int)
+                or not isinstance(cursor, int)
+                or limit <= 0
+                or cursor <= 0
+            ):
+                raise ValueError(
+                    "Both page_size and page_number must be positive integers"
```
Comment (petergarbers) [inline zep_python/message/client.py:82]: limit and cursor

File: zep_python/message/client.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        APIError
+            If unable to get messages for the session.
+        NotFoundError
+            If the session is not found.
+        """
+
+        if session_id is None or session_id.strip() == "":
+            raise ValueError("Session ID cannot be empty.")
+
+        url = f"/sessions/{session_id}/messages"
```
Comment (petergarbers) [inline zep_python/message/client.py:127]: include support ofr limit and cursor

File: zep_python/message/client.py [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+        try:
+            print(f"url: {url}")
```
Comment (petergarbers) [inline zep_python/message/client.py:130]: remove me

File: zep_python/message/client.py [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        if session_id is None or session_id.strip() == "":
+            raise ValueError("Session ID cannot be empty.")
+
+        if message_id is None or message_id.strip() == "":
+            raise ValueError("Message ID cannot be empty.")
+
+        url = f"/sessions/{session_id}/messages/{message_id}"
+
+        try:
+            print(f"url: {url}")
```
Comment (petergarbers) [inline zep_python/message/client.py:165]: remove me

File: examples/messages_async.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    user_id = uuid.uuid4().hex
+    user_request = CreateUserRequest(
+        user_id=user_id,
+        email="user@example.com",
+        first_name="John",
+        last_name="Doe",
+        metadata={"foo": "bar"},
+    )
+    try:
+        user = client.user.add(user_request)
```
Comment (petergarbers) [inline examples/messages_async.py:30]: async

File: examples/messages_async.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            f"Session Message not found for Session {session_id} and Message {message_id}. Got error: {e}"
+        )
+    except APIError as e:
+        print(
+            f"API error occurred while getting message for Session {session_id} and Message {message_id}. Got error: {e}"
+        )
+
+
+async def update_and_print_session_message_metadata(
+    client, session_id, first_session_message_id
```
Comment (petergarbers) [inline examples/messages_async.py:97]: message_id

File: examples/messages_sync.py [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    try:
+        result = client.memory.delete_memory(session_id)
+        print(f"Memory deleted: {result}")
+    except NotFoundError:
+        print("Memory not found for Session" + session_id)
+
+
+def main():
+    base_url = "http://localhost:8000"
+    api_key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.nEJCptN3CRfD_SQ4om4Oa2yh-ARPI41qkjj0La0Hw54"
```
[the remark this answers is not in the document]
Comment (petergarbers) [reply examples/messages_sync.py:124]: Put your key here

**Comment by danielchalef** [comment]
@petergarbers since the new message functionality is reliant on a new server version, we should update the following to `0.21.0`, which will be the next feature release version of Zep. Developers will be warned if using this client with an older version of the Zep server.

https://github.com/getzep/zep-python/blob/be81edcb749adfa4f5b1a436140fb1edc6d86bc6/zep_python/zep_client.py#L25

**Review by danielchalef — CHANGES_REQUESTED** [review]
see my comment regarding the minimum server version

**Review by danielchalef — APPROVED** [review]
lgtm

**Pull request merged**